pkg/reconciler/ksvc/resources/knative_configuration.go

```diff
@@ -0,0 +1,80 @@
+package resources
+
+import (
+	"fmt"
+	"github.com/knative/serving/pkg/apis/autoscaling"
+	knservingv1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha1"
+	"github.com/kubeflow/kfserving/pkg/constants"
+	"github.com/kubeflow/kfserving/pkg/frameworks/tensorflow"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func CreateModelServingContainer(modelName string, modelSpec *v1alpha1.ModelSpec) *v1.Container {
+	if modelSpec.Tensorflow != nil {
+		return tensorflow.CreateTensorflowContainer(modelName, modelSpec.Tensorflow)
+	} else {
+		//TODO(@yuzisun) handle other model types
+		return &v1.Container{}
+	}
+}
+
+func CreateKnativeConfiguration(kfsvc *v1alpha1.KFService) (*knservingv1alpha1.Configuration, *knservingv1alpha1.Configuration) {
```

Comment:
Consider separating logic between canary/default to make this a little cleaner.

Reply:
The reason I did this way is that configuration is composed of common fields like annotations, min/max replicas and default/canary spec fields, the purpose is to reuse the code setting common fields and default/canary fields are already encapsulated in `CreateModelServingContainer`.  Another issue I realize is that when user changes the common fields like min/max replicas, both default and canary configuration get applied, not sure if that's what user expects since user may want to canary min/max also ?

Reply:
I had originally thought min/max would be applied in a special way to canary/default, but now I'm feeling more like we push it a layer down. I cut an issue for this. Feel free to take this step and then revise in a future PR (I'm working on a proposal PR for this)

Reply:
sgtm